### PR TITLE
Introduce FEPointEvaluation::integrate_value()

### PR DESCRIPTION
**Pull request by peterrum**

### CODE REVIEW

File: include/deal.II/matrix_free/fe_point_evaluation.h [hunk 1]
```diff
@@ -86,6 +86,23 @@ namespace internal
         result[component] = vector_entry;
       }
 
+      static scalar_value_type
+      sum_value(const scalar_value_type &result)
```
Comment (peterrum) [inline include/deal.II/matrix_free/fe_point_evaluation.h:90]: @bergbauer I was not sure about the traits. Is there an easier way to accomplish the same?

File: include/deal.II/matrix_free/fe_point_evaluation.h [hunk 2]
```diff
@@ -916,6 +974,15 @@ class FEPointEvaluation
             const EvaluationFlags::EvaluationFlags &integration_flags,
             const bool                              sum_into_values = false);
 
+  /**
+   * Takes values at quadrature points, multiplies by the Jacobian determinant
+   * and quadrature weights (JxW) and sums the values for all quadrature
+   * points on the cell. The result is a scalar, representing the integral
+   * over the function over the cell.
```
Comment (mschreter) [inline include/deal.II/matrix_free/fe_point_evaluation.h:981]:
```suggestion
   * of the function over the cell.
```

File: tests/matrix_free/integrate_value_01.cc [hunk 3]
```diff
@@ -0,0 +1,315 @@
...
+{
+public:
+  RightHandSideFunction(const unsigned int n_components)
+    : Function<dim, Number>(n_components)
+  {}
+
+  Number
+  value(const Point<dim, Number> &p, const unsigned int component) const
+  {
+    return 1.0;
```
Comment (mschreter) [inline tests/matrix_free/integrate_value_01.cc:57]: If I am understanding it correctly, it always returns 1 --> is this on purpose?

**Comment by kronbichler** [comment]
Can you rebase this PR? If you do not want to address the comment now, I can pick that up in a separate PR, but I think we should not let this be around too long.

**Comment by peterrum** [comment]
@kronbichler I have rebased the branch and updated the comment.

**Review by kronbichler — APPROVED** [review]

**Pull request merged**